### PR TITLE
fix: issue with describe level test isolation and navigating to about:blank

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### Additional details

Fix issue with describe-level test isolation and navigating to about:blank. We need to ensure that we honor the *next* test's config when determining whether or not to navigate to about:blank at the end of the current test. This will allow an individual test to be isolated or not from previous tests

### Steps to test

The new system test specifies the scenario.

### How has the user experience changed?

n/a

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/driver/src/cypress/runner.ts [hunk 1]
```diff
@@ -74,10 +74,10 @@ const testBeforeRunAsync = (test, Cypress) => {
   })
 }
 
-const testBeforeAfterRunAsync = (test, Cypress) => {
+const testBeforeAfterRunAsync = (test, Cypress, additionalInfo) => {
```
Comment (mjhenkes) [inline packages/driver/src/cypress/runner.ts]: `additionalInfo` is a pretty lackluster argument name.

If you don't want to name the arg, you could do something like this i think.
```js
const testBeforeAfterRunAsync = (...args) => {
  const {test} = args
  return Promise.try(() => {
    if (!fired(TEST_BEFORE_AFTER_RUN_ASYNC_EVENT, test)) {
      return fire(TEST_BEFORE_AFTER_RUN_ASYNC_EVENT, ...args)
    }
  })
```

File: packages/driver/src/cypress/runner.ts [hunk 2]
```diff
@@ -497,6 +507,11 @@ const overrideRunnerHook = (Cypress, _runner, getTestById, getTest, setTest, get
         // If we're not in open mode or we're in open mode and not the last test we reset state.
         // The last test will needs to stay so that the user can see what the end result of the AUT was.
         if (shouldAlwaysResetPage || !lastTestThatWillRunInSuite(test, getAllSiblingTests(topSuite, getTestById))) {
+          const nextTest = nextTestThatWillRunInSuite(test, getAllSiblingTests(topSuite, getTestById))
+          const nextTestIsolationOverride = nextTest?._testConfig.unverifiedTestConfig.testIsolation
+          const topLevelTestIsolation = Cypress.originalConfig['testIsolation']
+          const nextTestHasTestIsolationOn = nextTestIsolationOverride || (nextTestIsolationOverride === undefined && topLevelTestIsolation)
```
Comment (mjhenkes) [inline packages/driver/src/cypress/runner.ts:513]:
```suggestion
          const nextTestHasTestIsolationOn = nextTestThatWillRunInSuite(test, getAllSiblingTests(topSuite, getTestById))?._testConfig.unverifiedTestConfig.testIsolation || (nextTestThatWillRunInSuite(test, getAllSiblingTests(topSuite, getTestById))?._testConfig.unverifiedTestConfig.testIsolation === undefined && Cypress.originalConfig['testIsolation'])
``` 

just kidding 😄

**Review by mjhenkes — APPROVED** [review]

File: packages/driver/src/cypress/runner.ts (hunk 2 above)
Comment (ryanthemanuel) [reply packages/driver/src/cypress/runner.ts:513]: Lol

File: packages/driver/src/cypress/runner.ts (hunk 1 above)
Comment (ryanthemanuel) [reply packages/driver/src/cypress/runner.ts]: Good call: [`ee5277a` (#27522)](https://github.com/cypress-io/cypress/pull/27522/commits/ee5277a17d303bad49e01a4addc8c0924e87735b)

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**